compiler/src/dotty/tools/dotc/transform/PostTyper.scala

```diff
@@ -157,7 +157,14 @@ class PostTyper extends MacroTransform with InfoTransformer { thisPhase =>
       val saved = inJavaAnnot
       inJavaAnnot = annot.symbol.is(JavaDefined)
       if (inJavaAnnot) checkValidJavaAnnotation(annot)
-      try transform(annot)
+      try
+        val res = transform(annot)
+        if res ne annot then
+          // If the annotation has been transformed, we need to make sure that the
```

Comment:
Why do we need to handle this here as well? I thought annotation transforms already took care of this themselves?